aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -313,11 +317,28 @@ func resourceAwsElasticBeanstalkEnvironmentUpdate(d *schema.ResourceData, meta i
 	envId := d.Id()
 
 	var hasChange bool
+	var hasTagChange bool
 
 	updateOpts := elasticbeanstalk.UpdateEnvironmentInput{
 		EnvironmentId: aws.String(envId),
 	}
 
+	updateTags := elasticbeanstalk.UpdateTagsForResourceInput{
+		ResourceArn: aws.String(d.Get("arn").(string)),
+	}
+
+	if d.HasChange("tags") {
```

Comment:
Can you move all the tag update logic inside this conditional here? We do not need to make this more complicated by spreading the logic everywhere like the other update call (which needs to build up arguments).